I'm testing this again, but there's a bit being left unsaid in this thread so I'm not sure I'm testing the right thing. Check my working and see where I'm going wrong.

First off, this branch doesn't evaluate as is due to infinite recursion.

```
$ git fetch origin pull/79793/head && git checkout FETCH_HEAD
[...]
$ nix-build pkgs/top-level/release-cross.nix -A bootstrapTools.armv7l.dist      
error: infinite recursion encountered, at undefined position
(use '--show-trace' to show detailed location information)
```

So I'm testing against the merge, which as of writing is:

```
$ git fetch origin pull/79793/merge && git checkout FETCH_HEAD
[...]
HEAD is now at 16c7ca5dd10 Merge 1f1b31c05afe88b45519d93a46b0bc9eafb5e490 into 816e214c1838d7e7265cdf1ffa361bfac852b38d

$ nix-build pkgs/top-level/release-cross.nix -A bootstrapTools.armv7l.dist
/nix/store/n3l34vzl1hdi5a7vs4rii4qgfh98zi8x-stdenv-bootstrap-tools-armv7l-unknown-linux-gnueabihf
```

I then plug the results into `pkgs/stdenv/linux/bootstrap-files/armv7l.nix` and `nix-build --argstr system armv7l-linux -A hello`. This builds for a while, then stops after
```
/nix/store/lmss9iwplnwcngv4xvhp4nsl9lkwh70n-bootstrap-tools/bin/ld: ../../src/liblzma/.libs/liblzma.so: undefined reference to `pthread_sigmask'
/nix/store/lmss9iwplnwcngv4xvhp4nsl9lkwh70n-bootstrap-tools/bin/ld: ../../src/liblzma/.libs/liblzma.so: undefined reference to `pthread_join'
/nix/store/lmss9iwplnwcngv4xvhp4nsl9lkwh70n-bootstrap-tools/bin/ld: ../../src/liblzma/.libs/liblzma.so: undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:501: xzdec] Error 1
make[3]: Leaving directory '/build/xz-5.2.4/src/xzdec'
make[2]: *** [Makefile:420: all-recursive] Error 1
make[2]: Leaving directory '/build/xz-5.2.4/src'
make[1]: *** [Makefile:613: all-recursive] Error 1
make[1]: Leaving directory '/build/xz-5.2.4'
make: *** [Makefile:480: all] Error 2
builder for '/nix/store/93qgcgmsyk5lidq8xx37488677ics6sj-xz-5.2.4.drv' failed with exit code 2

```

To reduce error I made a [quick script][] to glue the two halves together.

[quick script]: https://gist.github.com/thefloweringash/ec9511da8c7f41f903eb530a68f7e30f

--- a/pkgs/stdenv/linux/bootstrap-files/armv5tel.nix
+++ b/pkgs/stdenv/linux/bootstrap-files/armv5tel.nix
@@ -1,12 +1,12 @@
 {
   busybox = import <nix/fetchurl.nix> {
-    url = http://nixos-arm.dezgeg.me/bootstrap-2017-04-13-1f32d4b4/armv5tel/busybox;
-    sha256 = "00mxas5xg2j9n1g0q0nklr0dy87qqxk0jja5qz1yi7xl7zzsnpnw";
+    url = https://hydra.nixos.org/build/112609163/download/2/busybox;
+    sha256 = "0dc5471dc6a5f69ad98eb7445f51a61e88aa5792d7a677025bf012bdb513b763";
     executable = true;
   };
 
   bootstrapTools = import <nix/fetchurl.nix> {
-    url = http://nixos-arm.dezgeg.me/bootstrap-2017-04-13-1f32d4b4/armv5tel/bootstrap-tools.tar.xz;
-    sha256 = "0fhiy9l3mbmlhpkby31c2s63bhjiqx25qqr3wdp8cb7fxz8ayx2f";
+    url = https://hydra.nixos.org/build/112609163/download/1/bootstrap-tools.tar.xz;
+    sha256 = "ca0564eca4eb944649ce10ec70859640427bf2241243af62812b163176487e02";
   };
 }

--- a/pkgs/stdenv/linux/bootstrap-files/armv6l.nix
+++ b/pkgs/stdenv/linux/bootstrap-files/armv6l.nix
@@ -1,12 +1,12 @@
 {
   busybox = import <nix/fetchurl.nix> {
-    url = http://nixos-arm.dezgeg.me/bootstrap-2017-04-13-1f32d4b4/armv6l/busybox;
-    sha256 = "06n8dy8y2v28yx9ws8x64wxrvn9pszgpd299hc90nv9x21m79jzd";
+    url = https://hydra.nixos.org/build/112609441/download/2/busybox;
+    sha256 = "e6f6aecb675e924a96516f4379445dd2c0ba8b9c438fbfbaa2dc14ccce2802e0";
     executable = true;
   };
 
   bootstrapTools = import <nix/fetchurl.nix> {
-    url = http://nixos-arm.dezgeg.me/bootstrap-2017-04-13-1f32d4b4/armv6l/bootstrap-tools.tar.xz;
-    sha256 = "1gg2q3sw81vi65g1gmpvx0nnd6hxb76vlz73wfp292m90z1mym7f";
+    url = https://hydra.nixos.org/build/112609441/download/1/bootstrap-tools.tar.xz;
+    sha256 = "7a3f20def1a17ebf0edb5a92c403558429bcc2ac3d931b5e1bd88606cb217778";
   };
 }

--- a/pkgs/stdenv/linux/bootstrap-files/armv7l.nix
+++ b/pkgs/stdenv/linux/bootstrap-files/armv7l.nix
@@ -1,12 +1,12 @@
 {
   busybox = import <nix/fetchurl.nix> {
-    url = http://nixos-arm.dezgeg.me/bootstrap-2017-04-13-1f32d4b4/armv7l/busybox;
-    sha256 = "187xwzsng5lpak1nanrk88y4mlydmrbhx6la00rrd6kjx376s565";
+    url = https://hydra.nixos.org/build/112609103/download/2/busybox;
+    sha256 = "566a469dac214b31e4abdb0a91d32550bab1be5858d329e1b6074eef05370ca3";
     executable = true;
   };
 
   bootstrapTools = import <nix/fetchurl.nix> {
-    url = http://nixos-arm.dezgeg.me/bootstrap-2017-04-13-1f32d4b4/armv7l/bootstrap-tools.tar.xz;
-    sha256 = "05ayki2kak3i5lw97qidd5h9jv00dmlhx9h7l771bj331yamyqdn";
+    url = https://hydra.nixos.org/build/112609103/download/1/bootstrap-tools.tar.xz;
+    sha256 = "79fa2d7722aeb856c7c9b62a3fd64b6d261fd6f6bcbac486f0a2a7d823210550";
   };
 }

